/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

option java_multiple_files = true;
package org.apache.airavata.mft.resource.stubs.scp.storage;

message SCPStorage {
    string storageId = 1;
    string host = 2;
    int32 port = 3;
    string user = 4;
}

message SCPStorageGetRequest {
    string storageId = 1;
}

message SCPStorageCreateRequest {
    string host = 1;
    int32 port = 2;
    string user = 3;
}

message SCPStorageUpdateRequest {
    string storageId = 1;
    string host = 2;
    int32 port = 3;
    string user = 4;
}

message SCPStorageDeleteRequest {
    string storageId = 1;
}


